### PR TITLE
Add docs and Release notes info

### PR DESCRIPTION
**Pull request by kamilchodola**
Fixes Closes Resolves #

_Adding extended PR template to give devs a chance to put more info about PR and remind about docs._

## Changes

- _New fields added to PR template._

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [X] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [X] No

#### If yes, did you write tests?

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -35,6 +35,22 @@ _Please choose one of the keywords above to refer to the issue this PR solves fo
...
 _Optional. Remove if not applicable._
 
+## Documentation
+
+#### Requires documentation change?
+
+- [ ] Yes
+- [ ] No
+
+_Link to documentation or issue about change which needs to be added with "docs" label (example: https://github.com/NethermindEth/nethermind/issues/5856)_
```
Comment (rubo) [inline .github/pull_request_template.md:45]: This message should be more generic as if there's an issue for the PR, it's already mentioned at the very top and not necessarily should have the `docs` label. I'll commit the suggested change shortly.

**Review by rubo — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -37,19 +37,19 @@ _Optional. Remove if not applicable._
...
 ## Documentation
 
-#### Requires documentation change?
+#### Requires documentation update
 
 - [ ] Yes
 - [ ] No
 
-_Link to documentation or issue about change which needs to be added with "docs" label (example: https://github.com/NethermindEth/nethermind/issues/5856)_
+_If yes, fill in the details here or link the PR to docs update. Remove if not applicable._
```
Comment (kamilchodola) [inline .github/pull_request_template.md:45]: @rubo 
Thing with my "issue" approach was as discussed on call to create a separate issue with "docs" label and put it in appropriate project later to not forgot about this.
If you will just push details here and close PR, then those changes for sure will be forgotten.

Comment (kamilchodola) [reply .github/pull_request_template.md:45]: @rubo any opinion on above?

Comment (rubo) [reply .github/pull_request_template.md:45]: What about #5868?

**Review by rubo — APPROVED** [review]

**Review by benaadams — APPROVED** [review]

**Review by smartprogrammer93 — APPROVED** [review]

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**